File: Data Structures & Algorithms/permutations/submission-3.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res = []

        def dfs(cur_list, nums):
            if len(cur_list) == len(nums):
                res.append(cur_list[:])
                return
            
            for num in nums:
                if num in cur_list:
                    continue
                
                cur_list.append(num)
                dfs(cur_list, nums)
                cur_list.pop()
        
        dfs([], nums)
        return res

File: Data Structures & Algorithms/majority-element/submission-3.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut res = 0;
        let mut count = 0;

        for num in nums {
            if count == 0 {
                res = num
            }

            if num == res {
                count += 1
            } else {
                count -= 1
            }
        }
        res
    }
}
